securecookie.go

```diff
@@ -421,7 +347,9 @@ func decrypt(block cipher.Block, value []byte) ([]byte, error) {
 
 // Serialize encodes a value using gob.
 func (e GobEncoder) Serialize(src interface{}) ([]byte, error) {
-	buf := new(bytes.Buffer)
+	buf := getBuffer()
+	defer bufferPool.Put(buf)
```

Comment:
You're right—I'd considered that but had come to the wrong conclusion.

Given this, and how `Serialize` is called there likely isn't a straightforward way to use a buffer pool here. We can't change the method signature (public API) to return the buffer, and since we only have a `[]byte` we can't return that to the pool at the end of `Encode` - https://github.com/elithrar/securecookie/blob/pooled-buffers/securecookie.go#L189-L211 (unless I'm missing something).